kratos/solving_strategies/strategies/residualbased_newton_raphson_strategy.h

```diff
@@ -259,6 +259,7 @@ class ResidualBasedNewtonRaphsonStrategy
      
     virtual ~ResidualBasedNewtonRaphsonStrategy()
     {
+        Clear();
```

Comment:
@msandre I still can't see the problem with the ML preconditioner. I have tried to take a look into the trilinos ML code and they are apparently careful enough not to deallocate the rA. I would like to understand the issue better, because I suspect that this fix is hiding an error somewhere else. Do you have an example that failed without your fix?